terraform/20-app/cognito.tf

```diff
@@ -1,50 +1,64 @@
-module "cognito" {
-  source = "../modules/cognito"
-  sns_role_arn = aws_iam_role.cognito_sns_role.arn
-  user_pool_name    = "${local.prefix}-user-pool"
-  client_name       = "${local.prefix}-client"
-  user_pool_domain  = "${local.prefix}-domain"
-  callback_urls = concat(
-    ["https://${terraform.workspace}.dev.ukhsa-dashboard.data.gov.uk/api/auth/callback/cognito"],
-    local.is_dev ? ["http://localhost:3000/api/auth/callback/cognito", "http://localhost:3001/api/auth/callback/cognito"] : []
-  )
-  logout_urls = concat(
-    ["https://${terraform.workspace}.dev.ukhsa-dashboard.data.gov.uk"],
-    local.is_dev ? ["http://localhost:3000", "http://localhost:3001"] : []
-  )
-  region = local.region
-
-  ukhsa_oidc_client_id      = "ukhsa-oidc-client-id"
-  ukhsa_oidc_client_secret  = "ukhsa-oidc-client-secret"
-  ukhsa_oidc_issuer_url     = "https://example.com/issuer"
-  ukhsa_oidc_attributes_url = "https://example.com/attributes"
-
-  lambda_role_arn           = aws_iam_role.cognito_lambda_role.arn
-  prefix = local.prefix
+data "aws_secretsmanager_secret" "ukhsa_oidc_credentials" {
+  name = "${local.prefix}-ukhsa-oidc-credentials"
+}
+
+data "aws_secretsmanager_secret_version" "ukhsa_oidc_credentials" {
+  secret_id = try(data.aws_secretsmanager_secret.ukhsa_oidc_credentials.id, null)
+}
+
+data "aws_secretsmanager_secret" "ukhsa_tenant_id" {
+  name = "${local.prefix}-ukhsa-tenant-id"
+}
+
+data "aws_secretsmanager_secret_version" "ukhsa_tenant_id" {
+  secret_id = try(data.aws_secretsmanager_secret.ukhsa_tenant_id.id, null)
 }
 
-module "app_security_group" {
-  source  = "terraform-aws-modules/security-group/aws"
-  version = "~> 5.0"
+locals {
+  # Retrieve values from Secrets Manager
+  ukhsa_oidc_credentials = try(jsondecode(data.aws_secretsmanager_secret_version.ukhsa_oidc_credentials.secret_string), {})
+  ukhsa_tenant_id        = try(jsondecode(data.aws_secretsmanager_secret_version.ukhsa_tenant_id.secret_string)["tenant_id"], "")
 
-  name        = "${local.prefix}-security-group"
-  description = "Security group for the application"
-  vpc_id      = module.vpc.vpc_id
+  # Ensure correct retrieval of OIDC credentials
+  ukhsa_oidc_client_id     = lookup(local.ukhsa_oidc_credentials, "client_id", "")
+  ukhsa_oidc_client_secret = lookup(local.ukhsa_oidc_credentials, "client_secret", "")
+
+  # Define callback and logout URLs
+  default_callback_urls = ["https://${terraform.workspace}.dev.ukhsa-dashboard.data.gov.uk/api/auth/callback/cognito"]
```

Comment:
We'll definitely need to revisit these callback urls soon. Once we've got well known envs and basically anything that doesn't sit in the dev account.

Reply:
I’ve updated the callback URLs to be dynamically mapped based on `terraform.workspace`, ensuring we support well-known environments like test, uat and prod